Desktop_Interface/genericusbdriver.cpp

```diff
@@ -81,54 +81,53 @@ void genericUsbDriver::setFunctionGen(int channel, functionGenControl *fGenContr
        //// CH1 is AUX!!  CH2 is "MAIN"!!////
       //////////////////////////////////////
 
-    int length, maxLength, numDivides, maxDivides;
-    double freq, amplitude, offset;
-    unsigned char *samples;
+    int maxLength; // NOTE: unused
+    int maxDivides;
 
     //For recalling on crash.
-    if (channel == 0) fGenPtr_CH1 = fGenControl;
-    else fGenPtr_CH2 = fGenControl;
+    if (channel == 0)
+		fGenPtr_CH1 = fGenControl;
+    else
+		fGenPtr_CH2 = fGenControl;
 
     //Reading in data
-    if (channel == 0){
-        length = fGenControl->length_CH1;
-        freq = fGenControl->freq_CH1;
-        amplitude = fGenControl->amplitude_CH1;
-        offset = fGenControl->offset_CH1;
-        samples = (unsigned char *) malloc(length);
-        memcpy(samples, fGenControl->samples_CH1, (unsigned int) length);
-        numDivides = fGenControl->divisibility_CH1;
-    }
-    else if(channel == 1){
-        length = fGenControl->length_CH2;
-        freq = fGenControl->freq_CH2;
-        amplitude = fGenControl->amplitude_CH2;
-        offset = fGenControl->offset_CH2;
-        samples = (unsigned char *) malloc(length);
-        memcpy(samples, fGenControl->samples_CH2, (unsigned int) length);
-        numDivides = fGenControl->divisibility_CH2;
-    }
+	functionGenControl::ChannelData channelData = channel == 0
+		? fGenControl->CH1
+		: fGenControl->CH2;
+
+	// NOTE: Could be a copy constructor? Maybe not since this is the only usage...
+	{
+		unsigned char *originalSamples = channelData.samples;
+		channelData.samples = (unsigned char *) malloc(channelData.length);
+		memcpy(channelData.samples, originalSamples, (unsigned int) channelData.length);
+	}
 
     //Triple mode
-    if ((amplitude+offset) > FGEN_LIMIT){
-        amplitude = amplitude / 3;
-        offset = offset / 3;
+    if ((channelData.amplitude+channelData.offset) > FGEN_LIMIT){
+        channelData.amplitude = channelData.amplitude / 3;
+        channelData.offset = channelData.offset / 3;
         fGenTriple |= ((unsigned char) !channel + 1);
     }
     else fGenTriple &= ((unsigned char) (254 - !channel));
 
-    //qDebug() << "fGenTriple = " << fGenTriple << "fGenControl = " << fGenControl << "length = " << length << "freq = " << freq << "amplitude = " << amplitude << "offset = " << offset << "samples = " << samples;
+    // qDebug() << "fGenTriple = " << fGenTriple
```

Comment:
No point leaving in commented out print statements.  They should have been deleted years ago.